From b06b0ff8aa937b0345966e1b6c24757a10f727cb Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 18 Jan 2016 13:44:56 +0800
Subject: [PATCH 0855/5242] MLK-12202: ASoC: fsl_asrc_m2m: fix dma timeout
 after suspend/resume

commit  3c65f04560133f5d0690ee46b636b4ac9e3e31e8 from
https://source.codeaurora.org/external/imx/linux-imx.git

In suspend function, the complete will be set to done in callback.
After resuming, the convert will not spend time to wait the complete.
which is a wrong complete.
So in suspend function, the complete need to be reinited for next convert.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_asrc_m2m.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_asrc_m2m.c b/sound/soc/fsl/fsl_asrc_m2m.c
index 0caed62..0e5db3d 100644
--- a/sound/soc/fsl/fsl_asrc_m2m.c
+++ b/sound/soc/fsl/fsl_asrc_m2m.c
@@ -1,7 +1,7 @@
 /*
  * Freescale ASRC Memory to Memory (M2M) driver
  *
- * Copyright (C) 2014-2015 Freescale Semiconductor, Inc.
+ * Copyright (C) 2014-2016 Freescale Semiconductor, Inc.
  *
  * This file is licensed under the terms of the GNU General Public License
  * version 2. This program is licensed "as is" without any warranty of any
@@ -954,11 +954,13 @@ static void fsl_asrc_m2m_suspend(struct fsl_asrc *asrc_priv)
 			if (pair->dma_chan[IN])
 				dmaengine_terminate_all(pair->dma_chan[IN]);
 			fsl_asrc_input_dma_callback((void *)pair);
+			init_completion(&m2m->complete[IN]);
 		}
 		if (!completion_done(&m2m->complete[OUT])) {
 			if (pair->dma_chan[OUT])
 				dmaengine_terminate_all(pair->dma_chan[OUT]);
 			fsl_asrc_output_dma_callback((void *)pair);
+			init_completion(&m2m->complete[OUT]);
 		}
 
 		spin_unlock_irqrestore(&asrc_priv->lock, lock_flags);
-- 
1.7.9.5

